tests/unit/test_charm.py

```diff
@@ -1,326 +1,153 @@
+    assert out.unit_status.name == "blocked"
+    assert out.unit_status.message == "Missing S3 relation"
+
+
+@patch("models.S3ConnectionInfo.verify", return_value=True)
+def test_s3_relation_connection_ok(
+    _, tmp_path, history_server_ctx, history_server_container, s3_relation
+):
+    state = State(
+        relations=[s3_relation],
+        containers=[history_server_container],
+    )
+    out = history_server_ctx.run(s3_relation.changed_event, state)
+    assert out.unit_status.name == "active"
+
+    # Check containers modifications
+    assert len(out.get_container(CONTAINER).layers) == 2
+
+    spark_properties = parse_spark_properties(out, tmp_path)
+
+    # Assert one of the keys
+    assert "spark.hadoop.fs.s3a.endpoint" in spark_properties
+    assert (
+        spark_properties["spark.hadoop.fs.s3a.endpoint"] == s3_relation.remote_app_data["endpoint"]
+    )
+
+
+@patch("models.S3ConnectionInfo.verify", return_value=False)
+def test_s3_relation_connection_ko(
+    _, tmp_path, history_server_ctx, history_server_container, s3_relation
+):
+    state = State(
+        relations=[s3_relation],
+        containers=[history_server_container],
+    )
+    out = history_server_ctx.run(s3_relation.changed_event, state)
+    assert out.unit_status.name == "blocked"
+    assert out.unit_status.message == "Invalid S3 credentials"
+
+
+@patch("models.S3ConnectionInfo.verify", return_value=True)
+def test_s3_relation_broken(
+    _, history_server_ctx, history_server_container, s3_relation, tmp_path
+):
+    state = State(
+        relations=[s3_relation],
+        containers=[history_server_container],
+    )
+    out = history_server_ctx.run(
+        s3_relation.broken_event, history_server_ctx.run(s3_relation.changed_event, state)
+    )
+    assert out.unit_status.name == "blocked"
+    assert out.unit_status.message == "Missing S3 relation"
+
+    spark_properties = parse_spark_properties(out, tmp_path)
+
+    # Assert one of the keys
+    assert "spark.hadoop.fs.s3a.endpoint" not in spark_properties
+
+
+def test_ingress_relation_creation(
+    tmp_path,
+    history_server_ctx,
+    history_server_container,
+    ingress_relation,
+):
+    state = State(
+        leader=True,
+        relations=[ingress_relation],
+        containers=[history_server_container],
+    )
+    out = history_server_ctx.run(ingress_relation.changed_event, state)
+    assert out.unit_status.name == "blocked"
+    assert out.unit_status.message == "Missing S3 relation"
+
+
+@patch("models.S3ConnectionInfo.verify", return_value=True)
+def test_with_ingress(
+    _, tmp_path, history_server_ctx, history_server_container, ingress_relation, s3_relation
+):
+    state = State(
+        relations=[s3_relation, ingress_relation],
+        containers=[history_server_container],
+    )
+    out = history_server_ctx.run(ingress_relation.changed_event, state)
+
+    assert out.unit_status.name == "active"
```

Comment:
fyi, you can do `isinstance(out.unit_status, ActiveStatus)`.

Reply:
Uhm, this does not work. I actually tried and then got errors, so I replaced with the name and message instead. I could actually use the `out.unit_status == ActiveStatus('')`` you suggested in another comment, it is a bit more readable I agree